src/hotspot/share/opto/phaseX.cpp

```diff
@@ -1160,6 +1184,813 @@ bool PhaseIterGVN::verify_node_value(Node* n) {
+      return false;
+
+    // ConvI2LNode::Identity does
+    // convert I2L(L2I(x)) => x
+    //
+    // Investigate why this did not already happen during IGVN.
+    //
+    // Found with:
+    //   compiler/loopopts/superword/TestDependencyOffsets.java#vanilla-A
+    //   -XX:VerifyIterativeGVN=1110
+    case Op_ConvI2L:
+      return false;
+
+    // MaxNode::find_identity_operation
+    //  Finds patterns like Max(A, Max(A, B)) -> Max(A, B)
+    //  This can be a 2-hop search, so maybe notification is not
+    //  good enough.
+    //
+    // Found with:
+    //   compiler/codegen/TestBooleanVect.java
+    //   -XX:VerifyIterativeGVN=1110
+    case Op_MaxL:
+    case Op_MinL:
+    case Op_MaxI:
+    case Op_MinI:
+    case Op_MaxF:
+    case Op_MinF:
+    case Op_MaxHF:
+    case Op_MinHF:
+    case Op_MaxD:
+    case Op_MinD:
+      return false;
+
+
+    // AddINode::Identity
+    // Converts (x-y)+y to x
+    // Could be issue with notification
+    //
+    // Turns out AddL does the same.
+    //
+    // Found with:
+    //  compiler/c2/Test6792161.java
+    //  -ea -esa -XX:CompileThreshold=100 -XX:+UnlockExperimentalVMOptions -server -XX:-TieredCompilation -XX:+IgnoreUnrecognizedVMOptions -XX:VerifyIterativeGVN=1110
+    case Op_AddI:
+    case Op_AddL:
+      return false;
+
+    // AbsINode::Identity
+    // Not investigated yet.
+    case Op_AbsI:
+      return false;
+  }
+
+  if (n->is_Load()) {
+    // LoadNode::Identity tries to look for an earier store value via
+    // can_see_stored_value. I found an example where this led to
+    // an Allocation, where we could assume the value was still zero.
+    // So the LoadN can be replaced with a zerocon.
+    //
+    // Investigate why this was not already done during IGVN.
+    // A similar issue happens with Ideal.
+    //
+    // Found with:
+    //   java -XX:VerifyIterativeGVN=1000 -Xcomp --version
+    return false;
+  }
+
+  if (n->is_Store()) {
+    // StoreNode::Identity
+    // Not investigated, but found missing optimization for StoreI.
+    // Looks like a StoreI is replaced with an InitializeNode.
+    //
+    // Found with:
+    //   applications/ctw/modules/java_base_2.java
+    //   -ea -esa -XX:CompileThreshold=100 -XX:+UnlockExperimentalVMOptions -server -XX:-TieredCompilation -Djava.awt.headless=true -XX:+IgnoreUnrecognizedVMOptions -XX:VerifyIterativeGVN=1110
+    return false;
+  }
+
+  if (n->is_Vector()) {
+    // Found with tier1-3. Not investigated yet.
+    // The observed issue was with AndVNode::Identity
+    return false;
+  }
+
+  Node* i = n->Identity(this);
+  // If we cannot find any other Identity, we are happy.
+  if (i == n) {
+    verify_empty_worklist(n);
+    return false;
+  }
+
+  // We just saw a new Identity which was not found during IGVN.
+  tty->cr();
+  tty->print_cr("Missed Identity optimization:");
+  tty->print_cr("Old node:");
+  n->dump_bfs(1, nullptr, "");
+  tty->print_cr("New node:");
+  i->dump_bfs(1, nullptr, "");
```

Comment:
```suggestion
  // The verificatin just found a new Identity that was not found during IGVN.
  tty->cr();
  tty->print_cr("Missed Identity optimization:");
  tty->print_cr("Old node:");
  n->dump_bfs(1, nullptr, "");
  tty->print_cr("New node:");
  i->dump_bfs(1, nullptr, "");
```
The wording of the comment confused me a bit :sweat_smile: 

Also, perhaps taking the tty lock might be appropriate since you are printing a lot here? Or do we know that only verification is printing at this point?